### PR TITLE
INFINITY-2326/2327 Cleanup of some startup warnings, and other minor logging improvements

### PR DESCRIPTION
**Pull request by nickbp**
- Annotations and generic type on ConfigResource: Fixing some jersey warnings emitted on scheduler startup.
- Downgrade of EndpointResource logs: Produces noise on every query, which is only really useful for service developers anyway
- Other log fixes: Nitpicks.

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/api/ConfigResource.java [hunk 1]
```diff
@@ -26,14 +26,15 @@
  *
  * @param <T> The configuration type which is being stored by the framework.
  */
+@Singleton
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/api/ConfigResource.java]: Why's this needed?

Comment (benclarkwood) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/api/ConfigResource.java]: (More out of curiosity than anything else. I haven't ever needed this annotation on resources.)

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java [hunk 2]
```diff
@@ -209,15 +219,14 @@ public void slaveLost(SchedulerDriver driver, Protos.SlaveID agentId) {
     }
 
     @Override
-    public void executorLost(SchedulerDriver driver, Protos.ExecutorID executorId, Protos.SlaveID slaveId, int status) {
+    public void executorLost(SchedulerDriver driver, Protos.ExecutorID executorId, Protos.SlaveID agentId, int status) {
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java]: :) 👍

**Comment by benclarkwood** [comment]
(I already kicked the C* job)

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/api/ConfigResource.java (hunk 1 above)
Comment (nickbp) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/api/ConfigResource.java]: solely to fix `WARNING: A resource, class com.mesosphere.sdk.api.ConfigResource, implements provider interface [interface javax.ws.rs.ext.ContextResolver] but does not explicitly define the scope (@Singleton, @PerLookup). The resource class will be managed as singleton.`

**Pull request merged**